File: src/main/java/org/micoli/php/events/ConfigurationEvents.java
package org.micoli.php.events;

import com.intellij.util.messages.Topic;
import org.micoli.php.configuration.models.Configuration;

public interface ConfigurationEvents {
    Topic<ConfigurationEvents> CONFIGURATION_UPDATED =
            Topic.create("PHP Companion Configuration Events", ConfigurationEvents.class);

    void configurationLoaded(Configuration configuration);
}

File: src/main/kotlin/org/micoli/php/events/ConfigurationEvents.kt
package org.micoli.php.events

import com.intellij.util.messages.Topic
import org.micoli.php.configuration.models.Configuration

interface ConfigurationEvents {
    fun configurationLoaded(loadedConfiguration: Configuration)

    companion object {
        @JvmField
        val CONFIGURATION_UPDATED: Topic<ConfigurationEvents> =
            Topic.create<ConfigurationEvents>(
                "PHP Companion Configuration Events", ConfigurationEvents::class.java)
    }
}
